### PR TITLE
[JN-1164] surfacing withdrawn enrollee data

### PR DESCRIPTION
**Pull request by devonbush**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

This adds a table of withdrawn enrollees, linked of the participant list 

image

image

The idea is to show just enough to help troubleshoot, but not sending any unnecessary PII.  The email column is hidden by default.

This also does a driveby addition of a doc link to the enrollee withdrawal modal 
image

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

1. redeploy ApiAdminApp
2. go to https://localhost:3000/demo/studies/heartdemo/env/sandbox/participants
3. click "withdrawn"
4. confirm table appears
5. confirm you can see emails if you unhide that column

### CODE REVIEW

File: ui-admin/src/study/participants/enrolleeView/AdvancedOptions.tsx [hunk 1]
```diff
@@ -28,10 +29,14 @@ export default function AdvancedOptions({ enrollee, studyEnvContext }:
     <form onSubmit={e => e.preventDefault()}>
       <h3 className="h5">Withdraw enrollee: {enrollee.profile.givenName} {enrollee.profile.familyName}</h3>
       <div>contact email: {enrollee.profile.contactEmail}</div>
+      <div className="my-3">
+        <strong>Withdrawal is permanent!</strong>  Read more about the
+        <ZendeskLink doc={DocsKey.WITHDRAWAL}> withdrawl process</ZendeskLink>.
```
Comment (connorlbark) [inline ui-admin/src/study/participants/enrolleeView/AdvancedOptions.tsx]:
```suggestion
        <ZendeskLink doc={DocsKey.WITHDRAWAL}> withdrawal process</ZendeskLink>.
```

File: ui-admin/src/study/participants/participantList/ParticipantList.tsx [hunk 2]
```diff
@@ -75,7 +76,7 @@ function ParticipantList({ studyEnvContext }: {studyEnvContext: StudyEnvContextT
           </Button>
         </div>
       }
-
+      <div><Link to={`withdrawn`}>Withdrawn</Link></div>
```
Comment (connorlbark) [inline ui-admin/src/study/participants/participantList/ParticipantList.tsx:79]: should we put this under a superuser only flag until we figure out the right place for it?

File: ui-admin/src/study/participants/participantList/WithdrawnEnrolleeList.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    enableRowSelection: true,
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel()
+  })
+  return <div className="container-fluid px-4 py-2">
+    { renderPageHeader('Withdrawn enrollees') }
+    <NavBreadcrumb value={'withdrawnList'}>Withdrawn</NavBreadcrumb>
+    <FontAwesomeIcon icon={faInfoCircle}/> More information about the
+    <ZendeskLink doc={DocsKey.WITHDRAWAL}> withdrawl process</ZendeskLink>.
```
Comment (connorlbark) [inline ui-admin/src/study/participants/participantList/WithdrawnEnrolleeList.tsx]:
```suggestion
    <ZendeskLink doc={DocsKey.WITHDRAWAL}> withdrawal process</ZendeskLink>.
```

**Review by connorlbark — APPROVED** [review]
It might not be new from this PR, but I was getting weird behavior with the selector. I've also been having lots of strange vite things happening, too, so I wouldn't be entirely surprised if this was something going weird with vite.

https://github.com/user-attachments/assets/e9918e7e-fc4c-4461-a49d-179290e90ec6

Looks good otherwise, but I personally feel like it's getting kinda crowded in the participant list button space, so I vote for making it superuser only or moving it somewhere more hidden.

**Comment by devonbush** [comment]
Yeah, that selector bug is over a year old.  we have a ticket for it somewhere, when I last looked at it ~8 months ago it looked like a non-trivial fix

File: core/src/main/java/bio/terra/pearl/core/service/participant/WithdrawnEnrolleeService.java [hunk 4]
```diff
@@ -13,6 +13,7 @@
 import bio.terra.pearl.core.service.workflow.EnrollmentService;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.With;
```
Comment (MatthewBemis) [inline core/src/main/java/bio/terra/pearl/core/service/participant/WithdrawnEnrolleeService.java:16]:
```suggestion
```

File: ui-admin/src/study/participants/enrolleeView/AdvancedOptions.tsx [hunk 5]
```diff
@@ -7,6 +7,7 @@ import { useNavigate } from 'react-router-dom'
 import { doApiLoad } from 'api/api-utils'
 import { Button } from 'components/forms/Button'
 import { Enrollee } from '@juniper/ui-core'
+import { DocsKey, ZendeskLink } from '../../../util/zendeskUtils'
```
Comment (MatthewBemis) [inline ui-admin/src/study/participants/enrolleeView/AdvancedOptions.tsx]:
```suggestion
import { DocsKey, ZendeskLink } from 'util/zendeskUtils'
```

**Review by MatthewBemis — APPROVED** [review]
Looks good! Just a few import nits

File: ui-admin/src/study/participants/participantList/ParticipantList.tsx (hunk 2 above)
Comment (devonbush) [reply ui-admin/src/study/participants/participantList/ParticipantList.tsx:79]: this feature was explicitly requested by OurHealth -- they want to be able to see the list of withdrawn participants

**Pull request merged**